### PR TITLE
CLI: use hub's `create_commit`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR changes the method to open PRs in `pt-to-tf` to the permanent method defined by the hub -- `create_commit`. It also updates the commit description (it now supports line breaks 🎉 ) and adds a flag to add extra description (so I can programmatically tag the right HF maintainer in certain repos)

We can see an example PR [here](https://huggingface.co/joaogante/test_text/discussions/9) -- @Rocketknight1 confirms that the notification got to him!

After this PR gets merged, we can announce `pt-to-tf`, as it no longer depends on unreleased functionality 🚀

### CODE REVIEW

**Comment by gante** [comment]
Some CI runs have hub 0.7.0 cached, figuring how to best update it

File: src/transformers/commands/pt_to_tf.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 import numpy as np
 from datasets import load_dataset
 
-from huggingface_hub import Repository, upload_file
+from huggingface_hub import Repository, create_commit
+from huggingface_hub._commit_api import CommitOperationAdd
```
Comment (sgugger) [inline src/transformers/commands/pt_to_tf.py:23]: You'll need to gate those imports behind a version check: Transformers accepts any version of `huggingface_hub` from 0.1.0 and onward.

File: setup.py [hunk 2]
```diff
@@ -67,6 +67,7 @@
     you need to go back to main before executing this.
 """
 
+
```
Comment (gante) [inline setup.py:70]: This extra line triggers a cache miss in GHA CI, which fetches the updated hub version.

File: src/transformers/commands/pt_to_tf.py [hunk 3]
```diff
@@ -171,9 +190,7 @@ def get_image_inputs(self):
 
     def run(self):
         # Fetch remote data
-        # TODO: implement a solution to pull a specific PR/commit, so we can use this CLI to validate pushes.
```
Comment (sgugger) [inline src/transformers/commands/pt_to_tf.py:174]: And here you shoud raise an explicit error if the version of huggingface_hub is not enough (unless no PR will be made of course).

**Review by sgugger — APPROVED** [review]
Thanks for updating this command to the new code!

**Review by Rocketknight1 — APPROVED** [review]
LGTM!

**Pull request merged**